Wall Stream Layout Migration #71

yurabakhtin opened the pull request:
Issues: https://github.com/humhub/humhub-modules-polls/issues/71 and https://github.com/humhub/tasks/issues/30

luke- commented on widgets/views/form.php:
```diff
@@ -11,9 +11,16 @@
 
+<div class="contentForm_options" data-content-component="polls.Poll" style="margin-bottom:10px;margin-top:0">
+    <?= Html::activeTextInput($model,'question', [
+        'placeholder' => Yii::t('PollsModule.widgets_views_pollForm', 'Short question...'),
```
![image](https://user-images.githubusercontent.com/4736168/96696683-55745000-138b-11eb-9c8e-66a02512e3c1.png)

![image](https://user-images.githubusercontent.com/4736168/96696904-98362800-138b-11eb-9ae5-9763b7fdb2d7.png)

yurabakhtin replied:
done

luke- reviewed (changes requested):
Looks very good for me. Just a few little things with the create form.

buddh4 commented:
@yurabakhtin could you change the richtext layout to `LAYOUT_INLINE` according to https://github.com/humhub/humhub-modules-polls/issues/73

yurabakhtin commented:
@luke- @buddh4 Changes for new layout in commit https://github.com/humhub/humhub-modules-polls/pull/74/commits/5c18c4795c12db01dd0021414bf36dbdd68f2091:

> could you change the richtext layout to LAYOUT_INLINE according to #73

<img width="958" alt="preview_layout" src="https://user-images.githubusercontent.com/6995524/96978650-41fdec00-1527-11eb-9abd-18dae31617f5.png">

luke- commented:
For me it now works as expected.

However, from a UI point of view, I don't like the 4 text inputs placed directly below each other.  
![image](https://user-images.githubusercontent.com/4736168/97007641-fefd4200-1541-11eb-9ab4-e5182ef2712f.png)

Maybe we could:

- Add 1-2 more rows to the description field per Default
- Place a grey "?" in the beginning of each "Answer" text input (similar to the trash can)
- Add headline "Possible answers:" above the Answer section.

What do you think?

orangiene commented:
hi all,
here are three suggestions for the form fields. I
![06_Stream_Polls-input-fields](https://user-images.githubusercontent.com/62644500/97019115-34a92780-1550-11eb-9a3f-1f70330f0abc.jpg)
 think the simplest version is the first one. it's your choice :)

luke- commented:
@orangiene Thanks for the variants!

I don't want to change too much in this stage and also in comparison to the current design.

@yurabakhtin As first step, can you please try following points:
- Make description multi line
- Add blank line between Title&Description and the answer section
- Maybe: Add title above answers
- Maybe: Make checkboxes two column

yurabakhtin commented:
@luke- Please check:
<img width="959" alt="new_changes_ui_polls" src="https://user-images.githubusercontent.com/6995524/97176277-c3fb4880-17a5-11eb-9f01-10b4b4a34df5.png">

buddh4 commented:
Looks nice, only the margin between questions/description/answers is a bit to high. Maybe use `ActiveForm::field()` for question and description, which will wrap the inputs in `form-group` divs similar to the edit view.

yurabakhtin commented:
@buddh4 I tested the `ActiveForm::field()`, the `form-group` has `margin-bottom: 15px;` but we don't need this space when only field "Question" is visible. I.e. on page load when only "Question" is visible we should keep this field with `margin:0`, then on click to the "Question" field we display "Description" below with inline styles:

- first version was `margin:15px 0` for the "Description" field,
- but @luke- wrote "*Add blank line between Title&Description and the answer section*" and I changed that to `margin:30px 0`
- now after your comment I changed to `margin:20px 0 15px` and added same label style for "Answers", please check how it looks:
<img width="961" alt="poll_add_edit" src="https://user-images.githubusercontent.com/6995524/97279883-97e6d280-184c-11eb-8bb4-20f16ae9e525.png">

If we will use `ActiveForm::field()` for "Description" then style class `form-group` will adds only `margin-bottom:15px` but we also need `margin-top:~15-30px` because "Question" cannot have `margin-bottom`.

buddh4 commented:
@yurabakhtin So the last screenshot is the current state? I think the margin and labels look fine in this one.

yurabakhtin commented:
@buddh4 yes, this is current state right after my last changes in commit https://github.com/humhub/humhub-modules-polls/pull/74/commits/a8563b7c8ebfe5e778cf30fdb998d25c1bbd1c91
